### PR TITLE
Py3 setup

### PR DESCRIPTION
**Pull request by jburel**
* Update to use python3 and virtual env
* For convenience, we only use one virtualenv. Web is kept for convenience but it will no longer appears in the ome-doc i.e. we recommend server and web to be separated
Web installation doc is generated by omeroweb-install

### CODE REVIEW

File: linux/install_centos7_nginx.sh [hunk 1]
```diff
@@ -33,20 +34,24 @@ if [[ "$PGVER" =~ ^(pg94|pg95|pg96|pg10)$ ]]; then
...
 if [ "$WEBSESSION" = true ]; then
-	bash -eux step05_2_websessionconfig.sh
+	su - omero -c "VIRTUALENV=$VIRTUALENV bash -eux step05_2_websessionconfig.sh"
 fi
 
 #If you don't want to use the systemd scripts you can start OMERO manually:
-#su - omero -c "OMERO.server/bin/omero admin start"
-#su - omero -c "OMERO.server/bin/omero web start"
+#su - omero -c ". $VIRTUALENV/bin/activate;OMERODIR=/home/omero/OMERO.server omero admin start"
+#su - omero -c ". $VIRTUALENV/bin/activate;OMERODIR=/home/omero/OMERO.server omero web start"
```
Comment (joshmoore) [inline linux/install_centos7_nginx.sh]: Have you looked at having OMERODIR in settings.env? Also: Simon had had some luck with just setting PATH & VIRTUAL_ENV rather than needing to activate the virtualenv if that would simplify things.

File: linux/install_centos7_nginx.sh [hunk 2]
```diff
@@ -6,9 +6,10 @@ WEBSESSION=${WEBSESSION:-false}
 OMEROVER=${OMEROVER:-latest}
 PGVER=${PGVER:-pg10}
 ICEVER=${ICEVER:-ice36}
+VIRTUALENV=${VIRTUALENV:-/home/omero/omeroenv}
 
-source `dirname $0`/settings.env
-source `dirname $0`/settings-web.env
+. `dirname $0`/settings.env
```
Comment (joshmoore) [inline linux/install_centos7_nginx.sh:10]: Why the change from `source` to `.`?

File: linux/omero-web-systemd.service [hunk 3]
```diff
@@ -5,11 +5,12 @@ Requires=nginx.service
 [Service]
 User=omero
 Type=forking
-PIDFile=/home/omero/OMERO.server/var/django.pid
+#PIDFile=/home/omero/omero/var/django.pid
```
Comment (joshmoore) [inline linux/omero-web-systemd.service]: Remove rather than commenT?

File: linux/omero-web-systemd.service [hunk 4]
```diff
@@ -5,11 +5,12 @@ Requires=nginx.service
...
 Restart=on-failure
 RestartSec=10
-ExecStart=/home/omero/OMERO.server/bin/omero web start
-ExecStop=/home/omero/OMERO.server/bin/omero web stop
+Environment="OMERODIR=/home/omero/OMERO.server"
+ExecStart=/home/omero/omeroenv/bin/omero web start
```
Comment (joshmoore) [inline linux/omero-web-systemd.service]: The ansible roles install to `/opt/omero/{web,server}/venv3`. Should we start to unify?

File: linux/step01_centos7_deps.sh [hunk 5]
```diff
@@ -1,8 +1,8 @@
 #!/bin/bash
 
-yum -y install python-{pip,devel,virtualenv,yaml,jinja2,tables}
+yum -y install python3-{pip,devel,virtualenv,yaml,jinja2,tables}
 
 
 #start-web-dependencies
-yum -y install python-pillow numpy
+yum -y install python3-pillow numpy
```
Comment (joshmoore) [inline linux/step01_centos7_deps.sh]: Several of these will now come when pip installing.

File: linux/step01_centos7_ice_venv.sh [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+
+#start-ice-py
+# Create a virtual env and activate it
+VIRTUALENV=${VIRTUALENV:-/home/omero/omeroenv}
+python3 -mvenv $VIRTUALENV
+. $VIRTUALENV/bin/activate
+
+# Install the Ice Python binding
+pip3 install https://github.com/ome/zeroc-ice-py-centos7/releases/download/0.2.1/zeroc_ice-3.6.5-cp36-cp36m-linux_x86_64.whl
```
Comment (joshmoore) [inline linux/step01_centos7_ice_venv.sh]: The discussion on whether or not we ship statically linked openssl'd versions is still outstanding.

File: linux/step04_all_omero.sh [hunk 7]
```diff
@@ -25,23 +29,29 @@ fi
...
 
 #configure
-OMERO.server/bin/omero config set omero.data.dir "$OMERO_DATA_DIR"
-OMERO.server/bin/omero config set omero.db.name "$OMERO_DB_NAME"
-OMERO.server/bin/omero config set omero.db.user "$OMERO_DB_USER"
-OMERO.server/bin/omero config set omero.db.pass "$OMERO_DB_PASS"
+
+
+# Install omero-py
+pip3 install "omero-py>=5.6.dev4"
```
Comment (joshmoore) [inline linux/step04_all_omero.sh]: This can use with unifying with the decision for the docs re: `--pre`, etc.

**Review by joshmoore — COMMENTED** [review]
Various thoughts/comments on a first read.

File: linux/install_centos7_nginx.sh (hunk 2 above)
Comment (jburel) [reply linux/install_centos7_nginx.sh:10]: i had problem on debian

Comment (joshmoore) [reply linux/install_centos7_nginx.sh:10]: Understood. i.e. https://wiki.ubuntu.com/DashAsBinSh#source

File: linux/step04_all_omero.sh (hunk 7 above)
Comment (jburel) [reply linux/step04_all_omero.sh]: This will be changed. Same for omero.server

File: linux/omero-web-systemd.service (hunk 4 above)
Comment (jburel) [reply linux/omero-web-systemd.service]: We should probably unify to make the transition easier. It depends if the permissions is set to create it under "/opt/
Note that I have kept "web installation" in this repo mainly for internal use.
This does not appear in the doc since we use omeroweb-install in the doc

File: linux/step04_all_omero.sh (hunk 7 above)
Comment (jburel) [reply linux/step04_all_omero.sh]: Which decision. I might have missed that

Comment (joshmoore) [reply linux/step04_all_omero.sh]: There's an open question of how we suggest installing the packages across the board.

Comment (jburel) [reply linux/step04_all_omero.sh]: ok so this can be finalised when we come up with a decision
I understood that a decision had been made

**Comment by pwalczysko** [comment]
```
[root@b9e774350374 /]# PGSETUP_INITDB_OPTIONS=--encoding=UTF8 /usr/pgsql-10/bin/postgresql-10-setup initdb
Failed to get D-Bus connection: Operation not permitted
failed to find PGDATA setting in postgresql-10.service
```

**Comment by will-moore** [comment]
I see that https://py3-ci.openmicroscopy.org/jenkins/job/OMERO-docs/lastSuccessfulBuild/artifact/omero/_build/html/sysadmins/version-requirements.html lists Postgres 11 as the optimal (best tested etc) for OMERO 5.6, but this guide uses Postgres 10?

**Comment by will-moore** [comment]
Also we specify Java 1.8 in this guide, but requirements page lists Java 11

**Comment by will-moore** [comment]
Manually went through https://py3-ci.openmicroscopy.org/jenkins/job/OMERO-docs/lastSuccessfulBuild/artifact/omero/_build/html/sysadmins/unix/server-centos7-ice36.html on a clean CentOS 7 Docker image and all worked fine. Started server and imported an image. 👍

**Comment by will-moore** [comment]
Ooops - I realise I used docker CentOS7 with python 2 not 3 (following my old blog post http://will-moore.github.io/omero-install-testing-with-docker/).
How do I get an CentOS7 image with python3?

**Comment by will-moore** [comment]
Sorry - ignore that - I can see that the steps involved install of python3

**Comment by joshmoore** [comment]
Merging along with https://github.com/ome/ome-documentation/pull/2038#issuecomment-561349685
